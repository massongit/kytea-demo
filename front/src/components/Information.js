import React from "react"
import PropTypes from "prop-types"
import {Table} from "react-bootstrap"
import {FormattedNumber, intlShape} from "react-intl"

/**
 * 解析結果表示部
 */
class Information extends React.Component {
    static propTypes = {
        number: PropTypes.number.isRequired,
        word: PropTypes.string,
        pos: PropTypes.string.isRequired,
        pronunciation: PropTypes.shape({
            id: PropTypes.number,
            margin: PropTypes.number.isRequired,
            pronunciation: PropTypes.string.isRequired
        }),
        intl: intlShape.isRequired
    }

    render() {
        return (
            <Table striped bordered hover condensed>
                <thead>
                    <tr>
                        {
                            [
                                this.props.intl.formatMessage({
                                    id: "header.candidate"
                                }),
                                this.props.intl.formatMessage({
                                    id: "header.word"
                                }),
                                this.props.intl.formatMessage({
                                    id: "header.pos"
                                }),
                                this.props.intl.formatMessage({
                                    id: "header.pronunciation"
                                }),
                                this.props.intl.formatMessage({
                                    id: "header.margin"
                                })
                            ].map((header, i) => ( // 見出しを上下左右中央揃えで表示
                                <th
                                    key={i}
                                    className="text-center"
                                    style={{
                                        "verticalAlign": "middle"
                                    }}
                                >
                                    {header}
                                </th>
                            ))
                        }
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        {
                            [
                                this.props.intl.formatNumber(this.props.number),
                                this.props.word,
                                this.props.pos,
                                this.props.pronunciation.pronunciation
                            ].map((data, i) => (
                                <td
                                    className="text-center"
                                    key={i}
                                >
                                    {data}
                                </td>
                            ))
                        }
                        <td className="text-right">
                            <FormattedNumber
                                value={this.props.pronunciation.margin}
                            />
                        </td>
                    </tr>
                </tbody>
            </Table>
        )
    }
}

export default Information
