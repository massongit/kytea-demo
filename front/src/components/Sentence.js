import React from "react"
import PropTypes from "prop-types"
import {ButtonGroup} from "react-bootstrap"
import Word from "../containers/Word"

/**
 * 入力文をボタンのグループとして表示するComponent
 * @param words {array} 文
 * @returns {element} 入力文をボタンのグループとして表示するComponent
 */
const Sentence = ({words}) => (
    <ButtonGroup>
        {
            (() => {
                if (words !== undefined) {
                    return words.map((word, i) => (
                        <Word
                            key={i}
                            word={word.word}
                            pos={word.pos}
                            pronunciation={word.pronunciation}
                        />
                    ))
                }
            })()
        }
    </ButtonGroup>
)

Sentence.propTypes = {
    words: PropTypes.array
}

export default Sentence
