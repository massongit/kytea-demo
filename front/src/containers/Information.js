import Information from "../components/Information"
import {connect} from "react-redux"
import {injectIntl} from "react-intl"

const mapStateToProps = state => ({
    number: state.showPOSAndPronunciation.number,
    word: state.showPOSAndPronunciation.word,
    pos: state.showPOSAndPronunciation.pos,
    pronunciation: state.showPOSAndPronunciation.pronunciation
})

/**
 * 解析結果表示部のContainer
 */
export default injectIntl(connect(mapStateToProps)(Information))
