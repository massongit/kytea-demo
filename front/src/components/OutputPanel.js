import React from "react"
import PropTypes from "prop-types"
import PanelBody from "./PanelBody"
import Information from "../containers/Information"
import Sentence from "../containers/Sentence"
import {Panel, Row} from "react-bootstrap"
import {FormattedMessage} from "react-intl"

/**
 * 解析結果表示部のパネル
 */
class OutputPanel extends React.Component {
    static propTypes = {
        word: PropTypes.string
    }

    render() {
        return (
            <Row>
                <Panel>
                    <Panel.Heading>
                        <FormattedMessage id="title.result"/>
                    </Panel.Heading>
                    <PanelBody messageId="description.result">
                        <Sentence/>
                    </PanelBody>
                    {
                        (() => {
                            // 表示する単語が選択されていない場合には表示しない
                            if (this.props.word) {
                                return (<Information/>)
                            }
                        })()
                    }
                </Panel>
            </Row>
        )
    }
}

export default OutputPanel
