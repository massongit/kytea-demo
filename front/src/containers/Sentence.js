import {connect} from "react-redux"
import Sentence from "../components/Sentence"

const mapStateToProps = state => ({
    words: state.showSentence.words
})

/**
 * 入力文をボタンのグループとして表示するComponentのContainer
 */
export default connect(mapStateToProps)(Sentence)
