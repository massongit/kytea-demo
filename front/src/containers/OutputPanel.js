import {connect} from "react-redux"
import OutputPanel from "../components/OutputPanel"

const mapStateToProps = state => ({
    word: state.showPOSAndPronunciation.word
})

/**
 * 解析結果表示部のパネルのContainer
 */
export default connect(mapStateToProps)(OutputPanel)
