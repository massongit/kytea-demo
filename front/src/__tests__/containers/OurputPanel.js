import React from "react"
import OutputPanel from "../../containers/OutputPanel"
import Information from "../../containers/Information"
import {shallow} from "enzyme"
import {makeStoreShowPOSAndPronunciation, makeStoreShowSentence} from "./Sentence"

/**
 * テストの前処理
 * @param createStore Storeを生成する関数
 */
const beforeProcess = (createStore) => (
    shallow(
        <OutputPanel
            store={createStore()}
        />
    ).dive()
)

describe("containers/OutputPanel", () => {
    it("初期状態からStateが遷移した際に、Componentが正しく配置されている", () => {
        for (const f of [makeStoreShowSentence, makeStoreShowPOSAndPronunciation]) {
            expect(beforeProcess(f)).toMatchSnapshot()
        }
    })

    it("初期状態からStateが遷移した際に、子要素にInformationが含まれない", () => {
        for (const v of [[makeStoreShowSentence, false], [makeStoreShowPOSAndPronunciation, true]]) {
            expect(beforeProcess(v[0]).children().contains(<Information/>)).toEqual(v[1])
        }
    })
})
