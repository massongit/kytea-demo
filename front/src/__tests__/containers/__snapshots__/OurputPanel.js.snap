// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/OutputPanel/showPOSAndPronunciationState 初期状態からshowSentence, showPOSAndPronunciationとStateが遷移した際に、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutputPanel
    dispatch={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": [Function],
          }
    }
    word="野球"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "row",
      "children": <Uncontrolled(Panel)>
        <PanelHeading
                bsClass="panel"
                componentClass="div"
        >
                <FormattedMessage
                        id="title.result"
                        values={Object {}}
                />
        </PanelHeading>
        <PanelBody
                messageId="description.result"
        >
                <Connect(Sentence) />
        </PanelBody>
        <InjectIntl(Connect(Information)) />
</Uncontrolled(Panel)>,
      "componentClass": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PanelHeading
            bsClass="panel"
            componentClass="div"
>
            <FormattedMessage
                        id="title.result"
                        values={Object {}}
            />
</PanelHeading>,
          <PanelBody
            messageId="description.result"
>
            <Connect(Sentence) />
</PanelBody>,
          <InjectIntl(Connect(Information)) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": <FormattedMessage
              id="title.result"
              values={Object {}}
/>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "title.result",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(Sentence) />,
            "messageId": "description.result",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": <Uncontrolled(Panel)>
          <PanelHeading
                    bsClass="panel"
                    componentClass="div"
          >
                    <FormattedMessage
                              id="title.result"
                              values={Object {}}
                    />
          </PanelHeading>
          <PanelBody
                    messageId="description.result"
          >
                    <Connect(Sentence) />
          </PanelBody>
          <InjectIntl(Connect(Information)) />
</Uncontrolled(Panel)>,
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PanelHeading
              bsClass="panel"
              componentClass="div"
>
              <FormattedMessage
                            id="title.result"
                            values={Object {}}
              />
</PanelHeading>,
            <PanelBody
              messageId="description.result"
>
              <Connect(Sentence) />
</PanelBody>,
            <InjectIntl(Connect(Information)) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "panel",
              "children": <FormattedMessage
                id="title.result"
                values={Object {}}
/>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "title.result",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(Sentence) />,
              "messageId": "description.result",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`containers/OutputPanel/showSentenceState 初期状態からshowSentenceへStateが遷移した際に、Componentが正しく配置されている 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutputPanel
    dispatch={[Function]}
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": [Function],
          }
    }
    word=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "row",
      "children": <Uncontrolled(Panel)>
        <PanelHeading
                bsClass="panel"
                componentClass="div"
        >
                <FormattedMessage
                        id="title.result"
                        values={Object {}}
                />
        </PanelHeading>
        <PanelBody
                messageId="description.result"
        >
                <Connect(Sentence) />
        </PanelBody>
</Uncontrolled(Panel)>,
      "componentClass": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PanelHeading
            bsClass="panel"
            componentClass="div"
>
            <FormattedMessage
                        id="title.result"
                        values={Object {}}
            />
</PanelHeading>,
          <PanelBody
            messageId="description.result"
>
            <Connect(Sentence) />
</PanelBody>,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": <FormattedMessage
              id="title.result"
              values={Object {}}
/>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "title.result",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(Sentence) />,
            "messageId": "description.result",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": <Uncontrolled(Panel)>
          <PanelHeading
                    bsClass="panel"
                    componentClass="div"
          >
                    <FormattedMessage
                              id="title.result"
                              values={Object {}}
                    />
          </PanelHeading>
          <PanelBody
                    messageId="description.result"
          >
                    <Connect(Sentence) />
          </PanelBody>
</Uncontrolled(Panel)>,
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PanelHeading
              bsClass="panel"
              componentClass="div"
>
              <FormattedMessage
                            id="title.result"
                            values={Object {}}
              />
</PanelHeading>,
            <PanelBody
              messageId="description.result"
>
              <Connect(Sentence) />
</PanelBody>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "panel",
              "children": <FormattedMessage
                id="title.result"
                values={Object {}}
/>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "title.result",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(Sentence) />,
              "messageId": "description.result",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          undefined,
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
